src/test/javascript/spec/component/text-submission-assessment/text-assessment-area.component.spec.ts

```diff
@@ -43,12 +53,45 @@ describe('TextAssessmentAreaComponent', () => {
             TextBlockRef.new(),
         ];
 
-        for (let i = 0; i < textBlockRefs.length; i++) {
-            component.textBlockRefs = textBlockRefs.slice(0, i);
+        for (let i = 0; i < textBlockRefs2.length; i++) {
+            component.textBlockRefs = textBlockRefs2.slice(0, i);
             fixture.detectChanges();
 
             const all = fixture.debugElement.queryAll(By.directive(TextblockAssessmentCardComponent));
             expect(all.length).toBe(i);
         }
     });
+
+    it('should toggle on alt', () => {
+        const spyOnAlt = jest.spyOn(component, 'onAltToggle');
+        const eventMock = new KeyboardEvent('keydown', { key: 'Alt' });
+
+        component.onAltToggle(eventMock, false);
+        expect(spyOnAlt).toHaveBeenCalledTimes(1);
+        expect(component.autoTextBlockAssessment).toBe(false);
+    });
+
+    it('should add TextBlockRef if text block is added manually', () => {
+        component.textBlockRefs = textBlockRefs;
+        jest.spyOn(component.textBlockRefsAddedRemoved, 'emit');
+        const expectedLength = component.textBlockRefs.length + 1;
+
+        component.addTextBlockRef(TextBlockRef.new());
+        fixture.detectChanges();
+
+        expect(component.textBlockRefsAddedRemoved.emit).toHaveBeenCalledTimes(1);
+        expect(component.textBlockRefs).toHaveLength(expectedLength);
+    });
+
+    it('should remove TextBlockRef if text block is deleted', () => {
+        component.textBlockRefs = textBlockRefs;
+        jest.spyOn(component.textBlockRefsAddedRemoved, 'emit');
+        const expectedLength = component.textBlockRefs.length - 1;
+
+        component.removeTextBlockRef(component.textBlockRefs[0]);
+        fixture.detectChanges();
+
+        expect(component.textBlockRefsAddedRemoved.emit).toHaveBeenCalledTimes(1);
+        expect(component.textBlockRefs).toHaveLength(expectedLength);
+    });
```

Comment:
It seems like these two test cases access the same ``textBlockRef`` variable from line 23. Therefore, these test cases affect each other: If the first one runs, there are two refs in the array when test case 2 starts (as one is added by the first test case). However, if I run test case 2 separately, there is (of course) only one ref in the array. In your case it only works in both cases because your calculating the expected length of the array in lines 77 and 89.

Testcases should be able to run without affecting each other. Instead of renaming the variable in line 43 to ``textBlockRefs2``, I suggest that you just use a local array in each of these two test cases, just like it's done for the test case in line 43. Then, you can expect an absolute length instead of calculating it, and the test becomes idempotent (which at least the second one is currently not as it would fail as soon as the array becomes empty after the n-th run).